### ISSUE TITLE
📎 Implement `eslint/max-params` `clippy/too_many_arguments`

### ISSUE THREAD

**Issue by minht11** [biomejs/biome#2916]
### Description

Implement [eslint/max-params](https://eslint.org/docs/latest/rules/max-params) and [clippy/too_many_arguments](https://rust-lang.github.io/rust-clippy/master/#/too_many_arguments) restricting number of parameter function/constructor/method can have.

Example from eslint:
```ts
/*eslint max-params: ["error", 3]*/

function foo1 (bar, baz, qux, qxx) {
    doSomething();
}
```

By default Eslint allows 3 and Clippy 7 params. For Biome I would like it to be 4 as a middle ground, because its not that bad yet, for example function taking `rgba` values as separate params. But number is up to debate.

For this rule I would also like to provide user configurable number of arguments option, since especially people in old codebases it can be non trival to migrate and this rule can still be useful without forcing it to be disabled.

Name: `useMaxNumberOfArguments`.

**Comment by Conaclos** [comment]
Do we really need this rule? We should check if many users actually use that. Seems pretty useless to me.

**Comment by minht11** [comment]
Hmm you are prob right, I don't see much usage based on [github search](https://github.com/search?q=eslint%2Fmax-params&type=code&p=1) and since it is not enabled by default not a lot of users will be using it. We could implement very high number like clippy 7, but likely very few people would be helped by that.

I haven't done much research about it I just picked it up it from the pinned [Linter rules from other sources](https://github.com/biomejs/biome/discussions/3#top)(https://github.com/biomejs/biome/discussions/3) clippy section :D. 

Should that be removed or marked as won't implement there?

**Comment by Conaclos** [comment]
I can closer the issue as unplanned.

**Comment by jfurnish-athena** [comment]
@Conaclos, @minht11 , I would vote for a rule allowing for explicit, enforced parameter names, rather than implied, position-based parameters, in the spirit of https://dev.to/gajus/the-most-important-eslint-rule-max-params-349l 

While it may not be the most common choice today, I would like the tools to enforce a culture of **readable**, maintainable code, and something like 
`setColor(100,200,0)`
is less preferable than
`setColor({red:100, green:200, blue:0})`

**Comment by everscending** [comment]
^ +1

**Comment by jfurnish-athena** [comment]
@Conaclos , yeah that would be moving in the right direction! I wonder if you could do something like report functions that both : 

1. have more than 1 input param
2. do NOT use an interface

basically, trying to encourage folks to do something like 
```
export interface functionArgs {
  base: string;
  id: number;
  username: string;
}
export const someFunction = async (functionArgs: functionArgs): Promise<any> => {...
```

just trying to brainstorm here, and again the ultimate goal is to encourage coding practices where the function invocation in some distant file looks like the following, and it's easily readable/understand-able without having to check the original function definition
```
const response = await someFunction({
  base: 'whatever',
  id: 139,
  username: 'someone',
});
```

**Comment by yairEO** [comment]
Please consider reopening this... 

Nobody likes seeing the below, because some junior was too lazy to make the arguments as Object:

```js
foo(undefined, undefined, undefined, undefined, undefined, true)
```

**Comment by ematipico** [comment]
I think it's fair to implement this rule. While we aim for a more modern codebase, the "object param" pattern isn't always known, and it has drawbacks too. I think it's a good opportunity to teach something, and **it's very easy to implement**.

For example:

```js
fn(a, b, c);
fn2({ a, b, c })
```

They are different and can't always be interchangeable. We could provide an option to tweak the max number of arguments

**Comment by Jayllyz** [comment]
Hi, can i work on this one please? How should we name this rule and what should be the default max params and the severity of the rule ? Thanks

**Comment by dyc3** [comment]
I think `useMaxParams` would be good. Not sure about the severity though, maybe `warn`? We can always bikeshed it in the code review.
